infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -413,6 +404,15 @@ private void handleStartingPodStatus(CompletableFuture<Void> podRunningFuture, P
 
       for (ContainerStatus cs : status.getContainerStatuses()) {
         ContainerStateTerminated terminated = cs.getState().getTerminated();
+
+        if (terminated == null) {
+          ContainerStateWaiting waiting = cs.getState().getWaiting();
+          // we've caught the container waiting for a restart after a failure
+          if (waiting != null) {
+            terminated = cs.getLastState().getTerminated();
+          }
+        }
```

Comment:
Will this miss a restarting container if it has been restarted before we handle it? It might be useful to also check `restartCount`

Reply:
Good point. The latest version of the code takes the `restartCount` into account.